src/backend/PortabilityService.Functions/Analyze.cs

```diff
@@ -24,40 +24,41 @@ public static class Analyze
             [Inject]IStorage storage,
             ILogger log)
         {
+            var submissionId = Guid.NewGuid().ToString();
             var analyzeRequest = await DeserializeRequest(req.Content);
+
             if (analyzeRequest == null)
             {
-                log.LogError("invalid request");
+                log.LogError("Invalid request {SubmissionId}", submissionId);
                 return req.CreateResponse(HttpStatusCode.BadRequest);
             }
 
-            var submissionId = Guid.NewGuid().ToString();
-            log.LogInformation("Created submission id {SubmissionId}", submissionId);
-
             try
             {
+                log.LogInformation("Created submission id {SubmissionId}", submissionId);
+
                 var saved = await storage.SaveToBlobAsync(analyzeRequest, submissionId);
```

Comment:
I don't think the comments about `await` with `.ConfigureAwait(false)` were addressed. :(